File: src/main/java/org/jdc/template/log/CrashlyticsTree.java
package org.jdc.template.log;

import android.util.Log;

import timber.log.Timber;

public class CrashlyticsTree extends Timber.Tree {

    @Override
    protected boolean isLoggable(String tag, int priority) {
        return priority == Log.ERROR;
    }

    @Override
    protected void log(int priority, String tag, String message, Throwable t) {
        if (priority == Log.ERROR) {
            if (t != null) {
//                Crashlytics.log(1, "message", message);
//                Crashlytics.logException(t);
            } else {
//                Crashlytics.logException(new NonFatalCrashLogException(message));
            }
        }
    }

    public class NonFatalCrashLogException extends RuntimeException {
        public NonFatalCrashLogException(String message) {
            super(message);
        }
    }
}

File: src/main/kotlin/org/jdc/template/log/CrashlyticsTree.kt
package org.jdc.template.log

import android.util.Log

import timber.log.Timber

class CrashlyticsTree : Timber.Tree() {
    override fun isLoggable(tag: String?, priority: Int): Boolean {
        return priority == Log.ERROR
    }

    override fun log(priority: Int, tag: String?, message: String, t: Throwable?) {
        if (priority == Log.ERROR) {
            if (t != null) {
//                Crashlytics.log(1, "message", message)
//                Crashlytics.logException(t)
            } else {
//                Crashlytics.logException(new NonFatalCrashLogException(message))
            }
        }
    }

    inner class NonFatalCrashLogException(message: String) : RuntimeException(message)
}
